docs_app/src/app/custom-elements/migration-timeline/components/migration-timeline/missing-information/deprecation-item-form.component.ts

```diff
@@ -0,0 +1,180 @@
+      </mat-form-field>
+
+      <mat-expansion-panel [expanded]="withCodeExamples$ | async" (expandedChange)="withCodeExamples$.next($event)">
+        <mat-expansion-panel-header>
+          Code Examples
+        </mat-expansion-panel-header>
+
+        <mat-form-field class="full">
+          <mat-label>Code example before the deprecation:</mat-label>
+          <textarea matInput rows="7" [formControlName]="'exampleBefore'"></textarea>
+        </mat-form-field>
+
+        <mat-form-field class="full">
+          <mat-label>Code example after the deprecation:</mat-label>
+          <textarea matInput rows="7" [formControlName]="'exampleAfter'"></textarea>
+        </mat-form-field>
+      </mat-expansion-panel>
+      <br/>
+
+      <mat-form-field>
+        <mat-label>BreakingChangeMsg:</mat-label>
+        <input matInput placeholder="Message of related BreakingChange item" [formControlName]="'breakingChangeSubjectAction'">
+      </mat-form-field>
+
+      <mat-form-field>
+        <mat-label>BreakingChangeVersion:</mat-label>
+        <input matInput placeholder="Semver string of breaking version"
+          [formControlName]="'breakingChangeVersion'">
+      </mat-form-field>
+
+      <mat-form-field>
+        <mat-label>breakingChangeSubjectAction:</mat-label>
+        <input matInput placeholder="SubjectAction of related BreakingChange item" [formControlName]="'breakingChangeSubjectAction'">
+      </mat-form-field>
+
+    </form>
+  `,
+  styles: [`
+    .full {
+      width: 100%;
+    }
+  `],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DeprecationItemFormComponent extends State<{ crawledData: any }> {
+  crawledData$ = new ReplaySubject<any>(1);
+  withCodeExamples$ = new BehaviorSubject<boolean>(false);
+  subjectSymbols = SubjectSymbols;
+  defaultValue = '@TODO';
+  releaseForm = this.fb.group({
+    version: ['7.0.0-alpha.0', Validators.required],
+  });
+  deprecationForm = this.fb.group({
+    sourceLink: ['https://github.com/ReactiveX/rxjs/tree/7.0.0-alpha.0/src', Validators.required],
+    subject: [''],
+    subjectSymbol: [''],
+    subjectAction: ['deprecated'],
+    deprecationMsgCode: ['', Validators.required],
+    reason: ['', Validators.required],
+    implication: ['', Validators.required],
+    exampleBefore: [this.defaultValue],
+    exampleAfter: [this.defaultValue],
+    breakingChangeVersion: ['8.0.0', Validators.required],
+    breakingChangeMsg: ['', Validators.required],
+    breakingChangeSubjectAction: ['', Validators.required]
+  });
+
+  latestValidValuePrepared$: Observable<RawDeprecation> = combineLatest(
+    this.deprecationForm.valueChanges.pipe(startWith(this.deprecationForm.value)),
+    this.withCodeExamples$
+  ).pipe(
+    map(([rawDeprecation, withCodeExamples]: [any, RawDeprecation, boolean]) => {
+      const d: RawDeprecation = {...rawDeprecation};
+      d.itemType = 'deprecation';
+      if (!withCodeExamples) {
+        delete d.exampleAfter;
```

Comment:
I'd prefer an immutable approache to remove the properties.

Reply:
done